pyterrier/ltr.py

```diff
@@ -128,9 +128,9 @@ def fit(self, topics_and_results_Train, qrelsTrain, topics_and_results_Valid, qr
         kwargs = self.fit_kwargs
         self.learner.fit(
             np.stack(tr_res["features"].values), tr_res["label"].values, 
-            group=tr_res.groupby(["qid"]).count()["docno"].values, # we name group here for lightgbm compat. 
+            group=tr_res.groupby(["qid"])["docno"].count().to_numpy(), # we name group here for lightgbm compat. 
```

Comment:
If I properly understand how `.fit()` works, `tr_res` and `va_res` will need to be sorted by `qid` for this to work as expected, since `.groupby()` sorts by the key.

This isn't a degradation from the current implementation though.

Reply:
Hmm interesting point. `tr_res` and `va_res` are `pd.merge`d with the qrels which I think will force the ordering.